CMakeLists.txt

```diff
@@ -1559,6 +1560,25 @@ elseif(EMSCRIPTEN)
   CheckLibUnwind()
 
 elseif(UNIX AND NOT APPLE AND NOT RISCOS AND NOT HAIKU)
+
+  if(SDL_DLOPEN_NOTES)
+    set(CHECK_ELF_DLNOTES_SRC [==[
+    #ifndef __ELF__
+      ELF DL notes is only supported on ELF platforms
+    #endif
+    __attribute__ ((used,aligned(4),section(".note.dlopen"))) static const struct {
+      struct { int a; int b; int c; } hdr; char name[4]; __attribute__((aligned(4))) char json[24];
```

Comment:
If these are meant to be 4-byte quantities, shouldn't they be `int32_t` or `uint32_t` or something? (Although `int` is 4-bytes on every ELF platform that I know of)

Reply:
You're right, but this is a compile-time test to make sure the compiler understands the attributes (used, aligned, section).
The actual code used is in `src/dynapi/SDL_dynapi_dlopennotes.h`.